### ISSUE TITLE
Potential Data-Leakage Risk in GAT_v0.72.py – Target-Subject Samples Used for Training

### ISSUE THREAD

**Issue by dengzuxuan** [eeyhsong/Adaptor#4]
Hi @eeyhsong,
First of all, thank you for sharing the code and the well-organised repository—it's been really helpful for our group while exploring cross-subject EEG transfer.
While reproducing the results reported in the paper, I noticed that the file Adaptor/GAT_v0.72.py loads and actively uses the target-subject’s data during the training phase.According to the cross-subject BCI literature (and the definition adopted in paper), the training set must only contain labeled trials from source subjects, whereas the target subject’s data should appear exclusively in the test set:

> The training set comprised labeled data from all subjects’ training sessions, excluding the target subject. The testing set included all data from the target subject’s testing session.

Loading the target-subject features in GAT_v0.72.py (lines 686–700) violates this constraint and turns the experiment into an intra-subject (or at best mixed-subject) evaluation instead of a true cross-subject transfer task. 

```
(sour_feature, feature) = self.Feature_Extractor((sour_img, img))
# classifier
out_cls = self.Classifier(feature)
sour_out_cls = self.Classifier(sour_feature)
# discriminator
pre_cls_fake = self.Discriminator(sour_feature)
# Classification loss
loss_cls_targ = self.criterion_cls(out_cls, label)
# writer.add_scalar('Joint/cls_target', loss_cls_targ, e)
# writer.flush()
loss_cls_sour = self.criterion_cls(sour_out_cls, sour_label)
# writer.add_scalar('Joint/cls_source', loss_cls_sour, e)
# writer.flush()
loss_Joint_cls = loss_cls_targ + loss_cls_sour
```

Thanks again for the great work, and looking forward to your reply.
Best,
Colin
